File: Antibody.CareToKnowPro.CRM/Models/UserSpecialty.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Antibody.CareToKnowPro.CRM.Models
{
    public partial class UserSpecialty
    {
        public int UserSpecialtyId { get; set; }
        public int UserId { get; set; }
        public int SpecialityId { get; set; }
        [StringLength(200)]
        public string SpecialtyOther { get; set; }

        [ForeignKey("SpecialityId")]
        [InverseProperty("UserSpecialty")]
        public virtual Specialty Speciality { get; set; }
        [ForeignKey("UserId")]
        [InverseProperty("UserSpecialty")]
        public virtual User User { get; set; }
    }
}

File: Antibody.CareToKnowPro.CRM/ClientApp/src/app/model/userSpecialty.ts
import { ISpecialty as Specialty } from './specialty';
import { IUser as User } from './user';

export interface IUserSpecialty { 
    userSpecialtyId?: number;
    userId?: number;
    specialityId?: number;
    specialtyOther?: string;
    speciality?: Specialty;
    user?: User;
}
